Clean UI tests 4 of n

spencer3035 opened the pull request:
<!-- homu-ignore:start -->
<!--
If this PR is related to an unstable feature or an otherwise tracked effort,
please link to the relevant tracking issue here. If you don't know of a related
tracking issue or there are none, feel free to ignore this.

This PR will get automatically assigned to a reviewer. In case you would like
a specific user to review your work, you can assign it to them by using

    r? <reviewer name>
-->
<!-- homu-ignore:end -->
Cleaned up some tests that have `issue` in the title. I kept the commits to be one per "`issue`" cleanup/rename to make it easier to check. I can rebase to one commit once the changes are approved.

Related Issues:
#73494
#133895

r? jieyouxu

spencer3035 committed:
Cleaned up 4 tests in `tests/ui/issues`

jieyouxu commented on tests/ui/issues/issue-9719.rs:
```diff
@@ -1,40 +1,54 @@
+//! Ensure code generation when using polymorphic references does not trigger an LLVM assertion
```
Suggestion (wording): regression test for an LLVM assertion that used to be hit when:

- there's a generic enum contained within a tuple struct;
  - where the tuple struct is parameterized by some lifetime `'a`, and
  - the enum is concretized with its type argument being a reference to a trait object (of lifetime `'a`)

spencer3035 replied:
Reworded in ac2bab7a294077736127570a9587e2d62ca4e944

jieyouxu commented on tests/ui/argument-suggestions/exotic-calls.rs:
```diff
@@ -1,8 +1,17 @@
     //~^ ERROR function takes 0 arguments but 1 argument was supplied
 }
 
+fn foo2<T: Fn()>(f: T) {
+    |t| f(t);
+    //~^ ERROR function takes 0 arguments but 1 argument was supplied
+}
```
Suggestion: so I think this makes sense, however, can you remark that this specific case is a regression test for an ICE https://github.com/rust-lang/rust/issues/16939.

spencer3035 replied:
Added comment documenting issue number in 94017c04f1b0cb03bc2bacdd39b8f7c23f80b6f9

jieyouxu commented on tests/ui/warnings/no-dead-code-for-static-trait-impl.rs:
Question: can you double-check if there's a lint folder anywhere? Especially if there's a `dead_code/` folder somewhere in `tests/ui/`? This isn't really a "warning" but moreso a lint (`dead_code` specifically).

spencer3035 replied:
There's a folder `tests/ui/lint/dead-code` that I put this and the other one in. Moved in 4252767180a02931bf231c985609ad82193a3457 and a58d1afe8b330cdac084c2d5761b48e149ad5d54

jieyouxu commented on tests/ui/issues/issue-14421.rs:
```diff
@@ -1,15 +1,17 @@
+//! Issue: <https://github.com/rust-lang/rust/issues/14421>
 
+//@ check-pass
 //@ aux-build:issue-14421.rs
```
Question: I'm surprised that this doesn't fail in the next commit

jieyouxu replied:
Nvm it's fixed in the subsequent commit.

jieyouxu reviewed (approved):
Thanks, mostly looks good, a few nits. Please also squash the commits into one.

jieyouxu commented:
@rustbot author

spencer3035 commented:
Addressed comments. I'll wait till you resolve the discussions to squash to keep things clear.

Would you prefer in the future for me to just go ahead and rebase to remove the extra back-and-forth or do you prefer waiting to rebase until discussions are resolved?

@rustbot ready

jieyouxu commented:
> Would you prefer in the future for me to just go ahead and rebase to remove the extra back-and-forth or do you prefer waiting to rebase until discussions are resolved?

For the purposes of test changes like this PR, I don't think it really matters since we're going to squash anyway

jieyouxu reviewed (approved):
Thanks, I'll r+ after you squash the commits into one and PR CI is green.

jieyouxu commented:
@rustbot author

spencer3035 commented:
Squashed into one commit

@rustbot ready

jieyouxu commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 3eaa4b93684f10765ce9b8004fac461008ae678a has been approved by `jieyouxu`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=jieyouxu 3eaa4b93684f10765ce9b8004fac461008ae678a -->
<!-- homu: {"type":"Approved","sha":"3eaa4b93684f10765ce9b8004fac461008ae678a","approver":"jieyouxu","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
